src/antworld/world.cc

```diff
@@ -33,11 +33,22 @@ void readVector(std::istringstream &stream, std::vector<GLfloat> &vector, float
     }
 }
 //----------------------------------------------------------------------------
+template<unsigned int N, typename T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+void readVector(std::istringstream &stream, std::vector<T> &vector, float scale)
+{
+    // Read components and push back
+    GLfloat x;
+    for(unsigned int i = 0; i < N; i++) {
+        stream >> x;
+        vector.push_back((T)std::round(x * scale));
+    }
+}
+//----------------------------------------------------------------------------
 void readFace(std::istringstream &lineStream,
               const std::vector<GLfloat> &rawPositions,
-              const std::vector<GLfloat> &rawColours,
+              const std::vector<GLbyte> &rawColours,
               const std::vector<GLfloat> &rawTexCoords,
-              std::tuple<std::string, std::vector<GLfloat>, std::vector<GLfloat>, std::vector<GLfloat>> &currentObjSurface)
+              std::tuple<std::string, std::vector<GLfloat>, std::vector<GLbyte>, std::vector<GLfloat>> &currentObjSurface)
```

Comment:
I'm guessing that the ``std::vector<GLbyte>`` here contains sequences of 3 bytes for each colour channel?

Reply:
Also, maybe a comment is needed somewhere to describe what goes into these tuples.

Reply:
Added if you wanna merge!

Reply:
Cool